### PR TITLE
Fix Roslyn errors SA1512, CA1401, SA1633, SA1507, SA1309, SA1508, SA1210, SA1312, SA1111, SA1313

### PR DESCRIPTION
**Pull request by sanketshah11**
Fix Roslyn errors SA1512, CA1401, SA1633, SA1507, SA1309, SA1508, SA1210, SA1312, SA1111, SA1313

### CODE REVIEW

File: nuget/lib/BlingFireUtils.cs [hunk 1]
```diff
@@ -1,26 +1,40 @@
...
 using System.Runtime.InteropServices;
+using System.Text;
 
 /// <summary>
 /// This is C# interface for blingfiretokdll.dll
 /// </summary>
 namespace BlingFire
 {
-    public static class BlingFireUtils
+    public static class BlingFire
```
Comment (SergeiAlonichau) [inline nuget/lib/BlingFireUtils.cs:16]: We cannot change this name. It was already published.

Please ensure no public interface change.

File: nuget/lib/BlingFireUtils.cs [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
         private const string BlingFireTokDllName = "blingfiretokdll";
 
         [DllImport(BlingFireTokDllName)]
-        public static extern Int32 GetBlingFireTokVersion();
+        static extern Int32 GetBlingFireTokVersion();
+
+        public static Int32 GetBlingFireTokVersion(int fake = 0)
+        {
+            return GetBlingFireTokVersion();
+        }
```
Comment (SergeiAlonichau) [inline nuget/lib/BlingFireUtils.cs:26]: I am not sure this code works... did you run it? it looks like an infinite recursive call. Is there a way in C# to specify exactly which function to call? Other option is to just keep it simple and bite the bullet and add a public function GetVersion() which calls  private GetBlingFireTokVersion().

File: nuget/lib/BlingFireUtils.cs [hunk 3]
```diff
@@ -1,26 +1,40 @@
...
+        
 
         [DllImport(BlingFireTokDllName)]
-        public static extern UInt64 LoadModel(byte[] modelName);
+        static extern UInt64 LoadModel(byte[] modelName);
 
-        public static UInt64 LoadModel(string modelName)
```
Comment (SergeiAlonichau) [inline nuget/lib/BlingFireUtils.cs:21]: the public LoadModel should return long the types between LoadModel/FreeModel should match

File: nuget/lib/BlingFireUtils.cs [hunk 4]
```diff
@@ -101,262 +114,270 @@ public static IEnumerable<string> GetWords(string sentence)
...
+                MemoryMarshal.GetReference(inUtf8Str),
+                inUtf8StrLen,
+                ref MemoryMarshal.GetReference(outBuff),
+                maxBuffSize,
                 utf32SpaceCode);
         }
 
-        private static char[] _justNewLineChar = new char[] { '\n' };
-        private static char[] _justSpaceChar = new char[] { ' ' };
+        private static char[] justNewLineChar = new char[] { '\n' };
```
Comment (SergeiAlonichau) [inline nuget/lib/BlingFireUtils.cs:380]: Please rename it into g_justNewLineChar and g_justSpaceChar so it is clear it is static.

**Review by SergeiAlonichau — CHANGES_REQUESTED** [review]

File: nuget/lib/BlingFireUtils.cs (hunk 1 above)
Comment (sanketshah11) [reply nuget/lib/BlingFireUtils.cs:16]: My bad, will fix it.

File: nuget/lib/BlingFireUtils.cs (hunk 2 above)
Comment (sanketshah11) [reply nuget/lib/BlingFireUtils.cs:26]: reverted this change

File: nuget/lib/BlingFireUtils.cs (hunk 3 above)
Comment (sanketshah11) [reply nuget/lib/BlingFireUtils.cs:21]: reverted

File: nuget/lib/BlingFireUtils.cs (hunk 2 above)
Comment (sanketshah11) [reply nuget/lib/BlingFireUtils.cs]: I feel it is best to not make this change at this moment.

File: nuget/lib/BlingFireUtils.cs (hunk 3 above)
Comment (sanketshah11) [reply nuget/lib/BlingFireUtils.cs:21]: I feel it is best to not make this change at this moment.

**Review by SergeiAlonichau — APPROVED** [review]

**Pull request merged**